### PR TITLE
[Windows] move libdvd to cmake build tree

### PR DESCRIPTION
**Pull request by afedchin**
* allows to build libdvd for uwp
* builds x32/x64 ffmpeg with MSVC tools like on UWP
* drops mingw and simplify msys2

### CODE REVIEW

File: cmake/modules/FindLibDvd.cmake [hunk 1]
```diff
@@ -202,18 +202,81 @@ if(NOT WIN32)
...
     endif()
 else()
-  # Dynamically loaded on Windows
-  find_path(LIBDVD_INCLUDE_DIR dvdcss/dvdcss.h)
-
+  # windows specific builds
+  find_program(PATCH patch)
   include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(LibDvd REQUIRED_VARS LIBDVD_INCLUDE_DIR)
+  find_package_handle_standard_args(PATCH REQUIRED_VARS PATCH)
```
Comment (wsnipex) [inline cmake/modules/FindLibDvd.cmake]: this is not the correct place for searching a patch binary. Please move to windows depends

File: cmake/modules/FindLibDvd.cmake [hunk 2]
```diff
@@ -202,18 +202,81 @@ if(NOT WIN32)
...
+    file(STRINGS ${VERSION_FILE} VER)
+    string(REGEX MATCH "VERSION=[^ ]*$.*" ${dvdlib}_VER "${VER}")
+    list(GET ${dvdlib}_VER 0 ${dvdlib}_VER)
+    string(SUBSTRING "${${dvdlib}_VER}" 8 -1 ${dvdlib}_VER)
+    string(REGEX MATCH "BASE_URL=([^ ]*)" ${dvdlib}_BASE_URL "${VER}")
+    list(GET ${dvdlib}_BASE_URL 0 ${dvdlib}_BASE_URL)
+    string(SUBSTRING "${${dvdlib}_BASE_URL}" 9 -1 ${dvdlib}_BASE_URL)
+    string(TOUPPER ${dvdlib} DVDLIB)
+    set(${DVDLIB}_URL ${${dvdlib}_BASE_URL}/archive/${${dvdlib}_VER}.tar.gz)
+  endforeach()
```
Comment (wsnipex) [inline cmake/modules/FindLibDvd.cmake]: please don't duplicated this. 
The whole extra block for win shouldn't be needed and instead merged with non-depends build above.

**Review by wsnipex — CHANGES_REQUESTED** [review]

File: cmake/modules/FindLibDvd.cmake (hunk 2 above)
Comment (afedchin) [reply cmake/modules/FindLibDvd.cmake]: I did not get your thought

Comment (Rechi) [reply cmake/modules/FindLibDvd.cmake]: Above there is already the logic for building the libdvd libs for posix systems not using our depends.
You should integrate it there and only difference where it is really necessary (autoconf vs cmake, maybe also something else)

**Comment by afedchin** [comment]
jenkins build this please

File: cmake/modules/FindLibDvd.cmake (hunk 2 above)
Comment (afedchin) [reply cmake/modules/FindLibDvd.cmake]: done

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

File: cmake/modules/FindLibDvd.cmake [hunk 3]
```diff
@@ -1,133 +1,148 @@
...
-      core_link_library(${DVDNAV_LIBRARY} system/players/VideoPlayer/libdvdnav dvdnav archives "${_linklibs}")
-      set(LIBDVD_LIBRARIES ${DVDNAV_LIBRARY})
-      mark_as_advanced(LIBDVD_INCLUDE_DIRS LIBDVD_LIBRARIES)
+    core_link_library(${DVDNAV_LIBRARY} system/players/VideoPlayer/libdvdnav dvdnav archives "${_linklibs}")
+    set(LIBDVD_LIBRARIES ${DVDNAV_LIBRARY})
+    mark_as_advanced(LIBDVD_INCLUDE_DIRS LIBDVD_LIBRARIES)
+  endif()
+else()
+  set(dvdlibs libdvdread libdvdnav)
+  if(ENABLE_DVDCSS OR CORE_SYSTEM_NAME MATCHES windows)
```
Comment (Rechi) [inline cmake/modules/FindLibDvd.cmake]: Why explicitly enable for windows?

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
only wayland failed as usual

**Comment by afedchin** [comment]
@wsnipex @Rechi good to go?

**Review by wsnipex — APPROVED** [review]

**Review by Rechi — APPROVED** [review]

**Comment by Ae3NerdGod** [comment]
is this issue even being worked on?

**Pull request merged**